.travis.yml

```diff
@@ -1,106 +1,38 @@
-sudo: false
+# We will use docker to set up our environment, so don't use any particular
+# language in Travis itself
+language: generic
 
-language: cpp
+# Enable docker
+sudo: required
+services:
+    - docker
 
-os:
-  # - linux
-  - osx
+# Disable automatic submodule fetching ...
+git:
+    submodules: false
 
-osx_image: xcode8
+# ... in order to ensure we do a non-shallow submodule fetch
+before_install:
+    - git fetch --unshallow --tags
```

Comment:
I'll verify.  Probably the best way to handle the changelog issue is to tag the upstream `v0.10.0` release wth `v0.10.0.s0` (i.e. our version, vanilla copy of upstream) and to tag the patched version with `v0.10.0.s1` (meaning we have two well-defined annotated tags).

Upstream, it seems, is lax when it comes to decent tagging habits ;-)